### PR TITLE
Add new AaveLinearPool deployment

### PR DESCRIPTION
**Pull request by TomAFrench**
# Description

This PR adds a deployment task for the new AaveLinear pool with the fixes to the rebalancer's token approvals and protection for malicious reverts in external calls.

## Type of change

- [ ] Bug fix 
- [ ] New feature 
- [ ] Breaking change 
- [ ] Dependency changes
- [ ] Code refactor / cleanup
- [ ] Documentation or wording changes
- [x] Other

## Checklist:

- [x] The diff is legible and has no extraneous changes
- [x] Complex code has been commented, including external interfaces
- [x] Tests are included for all code paths
- [x] The base branch is either `master`, or there's a description of how to merge

### CODE REVIEW

File: pkg/deployments/README.md [hunk 1]
```diff
@@ -125,26 +125,27 @@ These deployments have been deprecated because they're either outdated and have
...
+| Fee Distributor for veBAL holders                       | [`20220420-fee-distributor`](./tasks/deprecated/20220420-fee-distributor)                           |
+| Mainnet Staking Gauges                                  | [`20220325-mainnet-gauge-factory`](./tasks/deprecated/20220325-mainnet-gauge-factory)               |
+| Arbitrum Root Gauges, for veBAL voting                  | [`20220413-arbitrum-root-gauge-factory`](./tasks/deprecated/20220413-arbitrum-root-gauge-factory)   |
+| Polygon Root Gauges, for veBAL voting                   | [`20220413-polygon-root-gauge-factory`](./tasks/deprecated/20220413-polygon-root-gauge-factory)     |
+| Optimism Root Gauges, for veBAL voting                  | [`20220628-optimism-root-gauge-factory`](./tasks/deprecated/20220628-optimism-root-gauge-factory)   |
+| Batch Relayer V3                                        | [`20220720-batch-relayer-v3`](./tasks/deprecated/20220720-batch-relayer-v3)                         |
+| Linear Pools for Aave aTokens with built-in rebalancing | [`20220817-aave-rebalanced-linear-pool`](./tasks/20220817-aave-rebalanced-linear-pool)              |
```
Comment (nventuro) [inline pkg/deployments/README.md:151]:
```suggestion
| Linear Pools for Aave aTokens with built-in rebalancing | [`20220817-aave-rebalanced-linear-pool`](./deprecated/tasks/20220817-aave-rebalanced-linear-pool)              |
```

(you'll need to adjust the formatting of the whole table)

File: pkg/deployments/tasks/20221115-aave-rebalanced-linear-pool/test/test.fork.ts [hunk 2]
```diff
@@ -38,7 +38,7 @@ describeForkTest('AaveLinearPoolFactory', 'mainnet', 15225000, function () {
   let poolId: string;
 
   before('run task', async () => {
-    task = new Task('20220817-aave-rebalanced-linear-pool', TaskMode.TEST, getForkedNetwork(hre));
+    task = new Task('20221115-aave-rebalanced-linear-pool', TaskMode.TEST, getForkedNetwork(hre));
```
Comment (nventuro) [inline pkg/deployments/tasks/20221115-aave-rebalanced-linear-pool/test/test.fork.ts:41]: Let's add a test where the rebalancer uses USDT. It'd also be great if we could have a simple test for a reverting rate, but I'm not sure we can easily craft that.

File: pkg/deployments/README.md [hunk 3]
```diff
@@ -104,6 +103,7 @@ Returns an object with all contracts from a deployment and their addresses.
 | Managed Pool                                            | [`20221021-managed-pool`](./tasks/20221021-managed-pool)                                             |
 | Protocol Fee Splitter                                   | [`20221027-fee-splitter`](./tasks/20221027-fee-splitter)                                             |
 | Authorizer Adaptor Entrypoint                           | [`20221111-authorizer-adaptor-entrypoint`](./tasks/20221111-authorizer-adaptor-entrypoint)           |
+| Linear Pools for Aave aTokens with built-in rebalancing | [`20221115-aave-rebalanced-linear-pool`](./tasks/20221115-aave-rebalanced-linear-pool)               |
```
Comment (nventuro) [inline pkg/deployments/README.md:106]: Add also a changelog entry

File: pkg/deployments/tasks/20221115-aave-rebalanced-linear-pool/readme.md [hunk 4]
```diff
@@ -0,0 +1,16 @@
+# 2022-11-15 - Aave Rebalanced Linear Pool
```
Comment (nventuro) [inline pkg/deployments/tasks/20221115-aave-rebalanced-linear-pool/readme.md:1]: The deprecated readme should have a deprecation notice and a link to this task (see other readmes for reference)

**Comment by TomAFrench** [comment]
> The build info I get on commit `362f269a93e84007db6a5c8151e588a01c880576` does not match what is committed here, are you sure you got the right one?

Hmm, forge-std might be mucking with the build info here. I'll make a fresh build-info.

File: pkg/deployments/CHANGELOG.md [hunk 5]
```diff
@@ -8,6 +8,7 @@
 - Deployed core infrastructure (`Authorizer`, `Vault`, `AuthorizerAdaptor`, `ProtocolFeeWithdrawer`, `ProtocolFeePercentagesProvider`, `BalancerQueries` and `BatchRelayer`) to Gnosis and BNB.
 - Deployed core Pool factories (`WeightedPoolFactory`, `ComposableStablePoolFactory`, `LiquidityBootstrappingPool`, `AaveLinearPool`) to BNB.
 - Deployed `AuthorizerAdaptorEntrypoint` to all networks.
+- Deployed `AaveLinearPool` to all networks.
```
Comment (nventuro) [inline pkg/deployments/CHANGELOG.md:11]:
```suggestion
- Deployed `AaveLinearPoolFactory` to all networks.
```

**Comment by TomAFrench** [comment]
> The build info I get on commit `362f269a93e84007db6a5c8151e588a01c880576` does not match what is committed here, are you sure you got the right one?

I've rebuild  from [c571f1a](https://github.com/balancer-labs/balancer-v2-monorepo/pull/2012/commits/c571f1a69091243618a245f277a6de5677e79cd5) and get the same build-info.

**Review by nventuro — COMMENTED** [review]
Been trying to get the build info check to pass, but it kept failing. According to @EndymionJkb, this could be because the file you submitted has had formatting applied to it, causing it to differ from the raw hardhat file. I tried a json diff tool and indeed I get a match when comparing them that way.

Could you update the build info so that it doesn't have this formatting applied to it? Thanks!

**Review by nventuro — APPROVED** [review]
I verified the build info matches what commit `1ae8c3c0d3a94337780bac29697fd7b0cd15d1a6` builds.

**Pull request merged**